src/msg/async/AsyncConnection.cc

```diff
@@ -362,14 +362,21 @@ ssize_t AsyncConnection::_try_send(bool more)
                              << " remaining bytes " << outcoming_bl.length() << dendl;
 
   if (!open_write && is_queued()) {
-    center->create_file_event(sd, EVENT_WRITABLE, write_handler);
-    open_write = true;
+    if (center->in_thread()) {
```

Comment:
@yuyuyu101 
That will very likely also fix my problem with Kqueue. Where I suspect that the handles are not kept when threads are forked. Hence the requirement to open an new Kqueue-handle.
But I've been side-tracked from this problem by the desire to first go to Cmake.
That is mostly done, so I'll soon return to looking into this